staging/src/k8s.io/apiserver/pkg/endpoints/discovery/version.go

```diff
@@ -29,12 +31,12 @@ import (
 )
 
 type APIResourceLister interface {
-	ListAPIResources() []metav1.APIResource
+	ListAPIResources() ([]metav1.APIResource, string)
 }
 
-type APIResourceListerFunc func() []metav1.APIResource
+type APIResourceListerFunc func() ([]metav1.APIResource, string)
 
-func (f APIResourceListerFunc) ListAPIResources() []metav1.APIResource {
+func (f APIResourceListerFunc) ListAPIResources() ([]metav1.APIResource, string) {
```

Comment:
Did you consider not taking the string but hashing the result of this function `[]metav1.APIResource` where the method is called? So that we don't have to change all the interfaces?

Reply:
Yes, but that would lead to unnecessary hashing - it would be impossible for the caller to know if the resource hasn't changed without marshalling/hashing it each time ServeHTTP is called